makefile circular dependency

Git clonning sceptre into fresh directory and running `make install-dev` will fail because the `clean` command fails because of `docs-clean` which fails because of `sphinx-apidoc` is not found:

```
$ make install-dev
rm -fr build/
rm -fr dist/
rm -fr .eggs/
find . -name '*.egg-info' -exec rm -fr {} +
find . -name '*.egg' -exec rm -f {} +
find . -name '*.pyc' -exec rm -f {} +
find . -name '*.pyo' -exec rm -f {} +
find . -name '*~' -exec rm -f {} +
find . -name '__pycache__' -exec rm -fr {} +
rm -fr .tox/
rm -fr .cache/
rm -f .coverage
rm -fr htmlcov/
rm -f test-results.xml
/Library/Developer/CommandLineTools/usr/bin/make -C docs clean
rm -f _api/sceptre.rst
rm -f _api/modules.rst
sphinx-apidoc -o _api ../sceptre
make[1]: sphinx-apidoc: No such file or directory
make[1]: *** [clean-api] Error 1
make: *** [docs-clean] Error 2
(sceptre)  ~/src/sceptre $ make install-dev
rm -fr build/
rm -fr dist/
rm -fr .eggs/
find . -name '*.egg-info' -exec rm -fr {} +
find . -name '*.egg' -exec rm -f {} +
find . -name '*.pyc' -exec rm -f {} +
find . -name '*.pyo' -exec rm -f {} +
find . -name '*~' -exec rm -f {} +
find . -name '__pycache__' -exec rm -fr {} +
rm -fr .tox/
rm -fr .cache/
rm -f .coverage
rm -fr htmlcov/
rm -f test-results.xml
/Library/Developer/CommandLineTools/usr/bin/make -C docs clean
rm -f _api/sceptre.rst
rm -f _api/modules.rst
sphinx-apidoc -o _api ../sceptre
make[1]: sphinx-apidoc: No such file or directory
make[1]: *** [clean-api] Error 1
make: *** [docs-clean] Error 2
(sceptre)  ~/src/sceptre $ make docs-clean
/Library/Developer/CommandLineTools/usr/bin/make -C docs clean
rm -f _api/sceptre.rst
rm -f _api/modules.rst
sphinx-apidoc -o _api ../sceptre
make[1]: sphinx-apidoc: No such file or directory
make[1]: *** [clean-api] Error 1
make: *** [docs-clean] Error 2
```

--- a/docs/Makefile
+++ b/docs/Makefile
@@ -20,3 +20,8 @@ help:
 
 apidoc:
 	@sphinx-apidoc -fM -o "$(SOURCEDIR)/apidoc" ../sceptre
+
+clean:
+	rm -f sceptre*rst
+	rm -f modules.rst
+	rm -rf _build

